File: Server/ChatServer/src/main/java/com/szlazakm/chatserver/dtos/OPKsCreateDTO.java
package com.szlazakm.chatserver.dtos;

import lombok.ToString;
import lombok.Value;
import lombok.experimental.SuperBuilder;
import lombok.extern.jackson.Jacksonized;

import java.util.List;

@Value
@SuperBuilder
@Jacksonized
@ToString
public class OPKsCreateDTO {

    List<OPKCreateDTO> opkCreateDTOs;
    String phoneNumber;
    Long nonceTimestamp;
    byte[] nonce;
    byte[] authMessageSignature;

}

File: client/app/src/main/java/com/szlazakm/safechat/webclient/dtos/OPKsCreateDTO.kt
package com.szlazakm.safechat.webclient.dtos

import com.fasterxml.jackson.annotation.JsonProperty

data class OPKsCreateDTO (
    @JsonProperty("phoneNumber")
    val phoneNumber: String,
    @JsonProperty("opkCreateDTOs")
    val opkCreateDTOs: List<OPKCreateDTO>,

    val nonceTimestamp: Long,
    val nonce: ByteArray,
    val authMessageSignature: ByteArray
)
